flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/BatchCompiledPlanTestBase.java

```diff
@@ -0,0 +1,262 @@
+            throws Exception {
+        final EnvironmentSettings settings = EnvironmentSettings.inBatchMode();
+        settings.getConfiguration().set(StateBackendOptions.STATE_BACKEND, "rocksdb");
+        final TableEnvironment tEnv = TableEnvironment.create(settings);
+        tEnv.getConfig()
+                .set(
+                        TableConfigOptions.PLAN_RESTORE_CATALOG_OBJECTS,
+                        TableConfigOptions.CatalogPlanRestore.IDENTIFIER);
+
+        program.getSetupConfigOptionTestSteps().forEach(s -> s.apply(tEnv));
+
+        for (SourceTestStep sourceTestStep : program.getSetupSourceTestSteps()) {
+            final Collection<Row> data = sourceTestStep.dataBeforeRestore;
+            final String id = TestValuesTableFactory.registerData(data);
+            final Map<String, String> options = new HashMap<>();
+            options.put("connector", "values");
+            options.put("data-id", id);
+            options.put("runtime-source", "NewSource");
+            options.put("terminating", "true");
+            options.put("bounded", "true");
+            sourceTestStep.apply(tEnv, options);
+        }
+
+        for (SinkTestStep sinkTestStep : program.getSetupSinkTestSteps()) {
+            final Map<String, String> options = new HashMap<>();
+            options.put("connector", "values");
+            options.put("disable-lookup", "true");
+            options.put("sink-insert-only", "false");
+            sinkTestStep.apply(tEnv, options);
+        }
+
+        program.getSetupFunctionTestSteps().forEach(s -> s.apply(tEnv));
+        program.getSetupTemporalFunctionTestSteps().forEach(s -> s.apply(tEnv));
+
+        final CompiledPlan compiledPlan =
+                tEnv.loadPlan(PlanReference.fromFile(getPlanPath(program, metadata)));
+
+        compiledPlan.execute().await();
+        for (SinkTestStep sinkTestStep : program.getSetupSinkTestSteps()) {
+            List<String> expectedResults = getExpectedResults(sinkTestStep, sinkTestStep.name);
+            assertThat(expectedResults)
+                    .containsExactlyInAnyOrder(
+                            sinkTestStep
+                                    .getExpectedBeforeRestoreAsStrings()
+                                    .toArray(new String[0]));
+        }
+    }
+
+    private Path getPlanPath(TableTestProgram program, ExecNodeMetadata metadata) {
+        return Paths.get(
+                getTestResourceDirectory(program, metadata) + "/plan/" + program.id + ".json");
+    }
+
+    private String getTestResourceDirectory(TableTestProgram program, ExecNodeMetadata metadata) {
+        return String.format(
+                "%s/src/test/resources/batch-compiled-plan-tests/%s/%d/%s",
+                System.getProperty("user.dir"), metadata.name(), metadata.version(), program.id);
+    }
+
+    private static List<String> getExpectedResults(SinkTestStep sinkTestStep, String tableName) {
+        if (sinkTestStep.getTestChangelogData()) {
```

Comment:
nit: this is a weird method name. it should rather be `shouldTestChangelogData`.

Reply:
Renaming.